jaxrs-api/src/main/java/javax/ws/rs/core/MediaType.java

```diff
@@ -367,7 +371,14 @@ public boolean equals(final Object obj) {
     @SuppressWarnings("UnnecessaryJavaDocLink")
     @Override
     public int hashCode() {
-        return (this.type.toLowerCase() + this.subtype.toLowerCase()).hashCode() + this.parameters.hashCode();
+        int h = this.hash;
+        if (h == 0) {
+            h = this.parameters.hashCode();
+            h = 31 * h + this.subtype.toLowerCase().hashCode();
+            h = 31 * h + this.type.toLowerCase().hashCode();
```

Comment:
DRY... You copied an algorithm implemented by the JRE. Instead of *copying* it, you simply should *invoke* `Object.hash(this.parameters, this.subtype, this.type)`.